### PR TITLE
constrain list actions to contexts with lists

### PR DESCRIPTION
**Pull request by frasertweedale**
Just a refactor I've wanted to do for a little while...

Changes:

```
a31b581 (Fraser Tweedale, 76 minutes ago)
   add purebred-email upper bound

   Ahead of anticipated breaking API changes in purebred-email, constrain
   purebred to <= 0.2.

deadd88 (Fraser Tweedale, 76 minutes ago)
   remove obsolete language extension

425c967 (Fraser Tweedale, 2 hours ago)
   constrain list actions to contexts with lists

   Add the HasList class and constrain list actions to contexts that have an
   instance of HasList.  This simplifies the implementation of current (or
   future) list actions.

   Also fix descriptions of the list actions.
```

### CODE REVIEW

**Comment by frasertweedale** [comment]
Guess I messed something up...

File: app/Main.hs [hunk 1]
```diff
@@ -1,4 +1,3 @@
-{-# LANGUAGE OverloadedStrings #-}
```
Comment (romanofski) [inline app/Main.hs]: How do you figure out that some are not in use? Just by knowing? hlint isn't telling you, right?

**Comment by frasertweedale** [comment]
Why yes, it would @romanofski.  Thanks for pointing that out.

I can't repro the UAT failure when running the test.  Do you have any comment?

File: app/Main.hs (hunk 1 above)
Comment (frasertweedale) [reply app/Main.hs]: hlint told me.  But also, this file is tiny and it's easy to see there is no string literal, let alone an overloaded one :)

**Comment by romanofski** [comment]
Let me have a look ...

**Comment by frasertweedale** [comment]
> Let me have a look ...

nevermind, I was only running the unit tests... :/

**Comment by frasertweedale** [comment]
@romanofski I have Questions.  I'll need to discuss this with you online I think.  It can wait.

**Pull request merged**